src/Type/ExponentiateHelper.php

```diff
@@ -105,4 +115,12 @@ private static function exponentiateConstantScalar(ConstantScalarType $base, Typ
 		return null;
 	}
 
+	/**
+	 * @return float|int
+	 */
+	private static function pow(mixed $base, mixed $exp)
```

Comment:
I had to wrap the pow operation into a separate method, as otherwise phpstan errored about invalid args to `pow()` / `**?